### PR TITLE
Update some doc about API reference.

### PR DESCRIPTION
**Pull request by qingqing01**
- Dropout
- L2Norm
- GRU
- Lstmp
- gaussian_random_batch_size_like
- random_data_generator
- box_coder
- save_combine

### CODE REVIEW

File: paddle/fluid/operators/detection/box_coder_op.cc [hunk 1]
```diff
@@ -106,23 +106,36 @@ class BoxCoderOpMaker : public framework::OpProtoAndCheckerMaker {
...
+
+    ow = exp(pwv * tw) * pw + tw / 2
+
+    oh = exp(phv * th) * ph + th / 2
+
+where `tx`, `ty`, `tw`, `th` denote the target box's center coordinates, width
+and height respectively. Similarly, `px`, `py`, `pw`, `ph` denote the
+priorbox's (anchor) center coordinates, width and height. `pxv`, `pyv`, `pwv`,
+`phv` denote the variance of the priorbox and `ox`, `oy`, `ow`, `oh` denote the
+encoded/decoded coordinates, width and height.
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/box_coder_op.cc:138]: image
image

File: paddle/fluid/operators/gaussian_random_batch_size_like_op.cc [hunk 2]
```diff
@@ -55,9 +56,11 @@ class GaussianRandomBatchSizeLikeOpMaker : public BatchSizeLikeOpMaker {
         .SetDefault(framework::proto::VarType::FP32);
 
     AddComment(R"DOC(
-GaussianRandom Operator.
 
 Used to initialize tensors with gaussian random generator.
+The defalut mean of the distribution is 0. and defalut standard
+deviation (std) of the distribution is 1.. Uers can set mean and std
+by input arguments.
```
Comment (qingqing01) [inline paddle/fluid/operators/gaussian_random_batch_size_like_op.cc:63]: image

File: python/paddle/fluid/layers/io.py [hunk 3]
```diff
@@ -386,16 +386,16 @@ def random_data_generator(low, high, shapes, lod_levels, for_parallel=True):
...
 
-         # Via the reader, we can use 'read_file' layer to get data:
-         image, label = fluid.layers.io.read_file(reader)
+            reader = fluid.layers.io.random_data_generator(
+                                             low=0.0,
+                                             high=1.0,
+                                             shapes=[(3,224,224), (1)],
+                                             lod_levels=[0, 0])
+            # Via the reader, we can use 'read_file' layer to get data:
+            image, label = fluid.layers.io.read_file(reader)
```
Comment (qingqing01) [inline python/paddle/fluid/layers/io.py]: image

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -624,17 +627,25 @@ def dynamic_gru(input,
...
+
         .. code-block:: python
 
+            dict_dim, emb_dim = 128, 64
+            data = fluid.layers.data(name='sequence', shape=(128),
+                                     dtype='int32', lod_level=1)
+            emb = fluid.layers.embedding(input=data, size=[dict_dim, emb_dim])
             hidden_dim = 512
-            x = fluid.layers.fc(input=data, size=hidden_dim * 3)
+            x = fluid.layers.fc(input=emb, size=hidden_dim * 3)
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py:700]: image
image

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -2532,32 +2544,33 @@ def l2_normalize(x, axis, epsilon=1e-12, name=None):
...
         .. code-block:: python
 
-          data = fluid.layers.data(name="data",
-                                   shape=(3, 17, 13),
-                                   dtype="float32")
-          normed = fluid.layers.l2_normalize(x=data, axis=1)
+            data = fluid.layers.data(name="data",
+                                     shape=(3, 17, 13),
+                                     dtype="float32")
+            normed = fluid.layers.l2_normalize(x=data, axis=1)
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py:3031]: image

File: python/paddle/fluid/layers/tensor.py [hunk 6]
```diff
@@ -486,11 +486,27 @@ def save_combine(x, file_path, overwrite=True):
...
+
+        .. code-block:: python
+
+            v1 = fluid.layers.data(name="data",
+                                   shape=(4, 6),
+                                   dtype="float32")
+            v2 = fluid.layers.data(name="data",
+                                   shape=(6, 8, 4),
+                                   dtype="float32")
+            normed = fluid.layers.save_combine([v1, v2], file_path="output")
```
Comment (qingqing01) [inline python/paddle/fluid/layers/tensor.py:520]: 该op在内部用，感觉用户不太需要，就没有暴露出来。

File: python/paddle/fluid/layers/nn.py [hunk 7]
```diff
@@ -888,13 +899,14 @@ def dropout(x, dropout_prob, is_test=False, seed=None, name=None):
...
+        Variable: A tensor variable is the shape with `x`.
 
     Examples:
+
         .. code-block:: python
 
-          x = fluid.layers.data(name="data", shape=[32, 32], dtype="float32")
-          droped = fluid.layers.dropout(input=x, dropout_rate=0.5)
+            x = fluid.layers.data(name="data", shape=[32, 32], dtype="float32")
+            droped = fluid.layers.dropout(input=x, dropout_rate=0.5)
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: image

**Comment by reyoung** [comment]
default
default

**Comment by qingqing01** [comment]
@reyoung Done.

- logsigmoid
image

- box_coder
image

**Review by reyoung — APPROVED** [review]
Excellent!

**Pull request merged**